bionic/executor.py

```diff
@@ -55,58 +26,74 @@ class Executor:
 
     def __init__(self, worker_count):
         self.worker_count = worker_count
-
-        class MyManager(SyncManager):
-            pass
-
-        MyManager.register("SynchronizedSet", SynchronizedSet)
-        self.process_manager = MyManager()
-        self.process_manager.start()
-
-        self._logging_queue = self.process_manager.Queue(-1)
-        self._logging_receiver = LoggingReceiver(self._logging_queue)
-        self._logging_receiver.start()
-
+        self._manager = get_singleton_manager()
         self._process_pool_exec = None
-        self.init_or_resize_process_pool(worker_count)
+        self.init_or_resize_process_pool()
 
-    def init_or_resize_process_pool(self, worker_count):
+    def init_or_resize_process_pool(self):
         """
-        If the process pool is not initialized yet, initializes it and
-        returns True.
-
-        If already initialized, resizes the process pool if the
-        worker_count is different than the worker count used previously
-        and returns whether the process pool was resized.
+        Initializes the process pool if not initialized yet. If  already initialized,
+        since the process pool is singleton in Loky, it attempts to register it again
+        which resizes the executor if the worker count changed.
         """
 
-        # When already initialized and worker count didn't change,
-        # no need to resize the process pool.
-        if self._process_pool_exec is not None and worker_count == self.worker_count:
-            return False
-
-        self.worker_count = worker_count
-
         # Loky uses cloudpickle by default. We should investigate further what would
         # it take to make is use pickle and wrap the functions that are non-picklable
         # using `loky.wrap_non_picklable_objects`.
         loky = import_optional_dependency("loky", purpose="parallel processing")
 
         self._process_pool_exec = loky.get_reusable_executor(
```

Comment:
Just to confirm: this is very cheap if the pool is not being resized, right? It might be worth stating this in a comment, because we're going to be calling this function a lot.

Reply:
Right, this is very cheap. Loky compares the original and new arguments to `get_reusable_executor` and simply returns the executor.